tests/clusters/fake_cluster.py

```diff
@@ -0,0 +1,102 @@
+from typing import Any, List, Mapping, MutableMapping, Optional, Sequence, Set, Tuple
+
+from snuba.clickhouse.native import ClickhousePool, Params
+from snuba.clusters.cluster import (
+    ClickhouseClientSettings,
+    ClickhouseCluster,
+    ClickhouseNode,
+)
+
+
+class FakeClickhousePool(ClickhousePool):
```

Comment:
I need to test with a fake cluster because, while we can run tests in cluster mode with distributed tables, everything has to be on the same node. I cannot start a cluster and access each node individually from outside the cluster.

Reply:
I think our distributed test should eventually have 2 shards at least as we skip testing quite a bit of functionality without it. If that were the case now, do you mean you use the real cluster instead of a fake one?

Reply:
If we had a real multi-node cluster such that snuba was capable to address each node yes. But that will require some docker tricks.
If you start the cluster with docker compose, the nodes can address each other inside the multi container applicaiton, this is good for clickhouse but then, in order to address individual nodes there are issues if snuba does not run in the same docker network. You can map individual docker nodes to localhost on a different port each, but then the port and host name you would find in the system.clusters table would not match with the host name you have to use outside of the docker compose application.

There should be a way to proxy connections from outside the docker compose application to the individual nodes inside, but I did not dig deeper.